lib/filewatch/read_mode/handlers/read_zip_file.rb

```diff
@@ -59,5 +68,27 @@ def close_and_ignore_ioexception(closeable)
         logger.warn("Ignoring an IOException when closing an instance of #{closeable.class.name}", "exception" => e)
       end
     end
+
+    def corrupted?(watched_file)
+      begin
+        file_stream = FileInputStream.new(watched_file.path)
+        gzip_stream = GZIPInputStream.new(file_stream)
+        buffer = Java::byte[8192].new
+        start = Time.new
+        until gzip_stream.read(buffer) == -1
+        end
+        return false
+      rescue ZipException => e
+        if e.message == 'invalid bit length repeat'
```

Comment:
this seems to be fine tuned to handle a particular case, not sure it covers possible corrupted states.
e.g. could also reach a `java.io.EOFException: Unexpected end of ZLIB input stream` prematurely.

maybe we should reverse the condition?  
find cases where `IOException`/`ZipException` does not indicate a corrupted file (makes sense to be re-raised), wdyt?

Reply:
An IOException could be a temporaly network connection error in NFS (also if heavily discourage to use NAS volumes with input-file)

Reply:
right, there's panoply of things going wrong - the reverse is not really an option.
we will need to try detecting corrupted archives case-by-case as you did here ...

Reply:
In general I agree with that that condition can't capture all broken gzip conditions. I'll try to improve it

Reply:
I've avoid the if condition on message, now every ZipException is identified as file corruption